### PR TITLE
Add recency and tf-idf weighted edges for project-focused graphs

### PR DESCRIPTION
**Pull request by christinaexyou**
@JamesKunstle Addresses #216 by:

* Adding a helper function to calculate recency weights
* Adding a helper function to calculate contributor tf-idf

Let me know if you want to see these in a separate nb and/or applied to a visualization, especially contributor tf-idf.

cc: @hemajv @oindrillac

### CODE REVIEW

File: notebooks/graph_analysis/approaches/graph_helper_functions.ipynb [hunk 1]
```diff
@@ -16,6 +16,7 @@
    "outputs": [],
```
Comment (JamesKunstle) [inline notebooks/graph_analysis/approaches/graph_helper_functions.ipynb:16]: <pre class="ql-syntax" spellcheck="false">Line #9.    
</pre><p>make sure to copy the dataframe before you make any modifications. df's are passed by reference so changes within the function scope apply to the object outside of the scope as well.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/oss-aspen/Rappel/pull/236/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (JamesKunstle) [inline notebooks/graph_analysis/approaches/graph_helper_functions.ipynb:16]: <pre class="ql-syntax" spellcheck="false">Line #18.        # convert values in the created_at column from strings to datetime objects
</pre><p>copy df before modification</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/oss-aspen/Rappel/pull/236/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (JamesKunstle) [inline notebooks/graph_analysis/approaches/graph_helper_functions.ipynb:16]: <pre class="ql-syntax" spellcheck="false">Line #36.        tf = df.apply(lambda row: row.num_cntrbs / contributonsByRepo[row.repo_id], axis=1)
</pre><p>This definition of tf/idf is unclear to me. </p><p> </p><p>Could you comment the rationale of this definition in the docstring of the function? I'd like to know what Oindrilla and Hema think. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/oss-aspen/Rappel/pull/236/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by christinaexyou** [comment]
@JamesKunstle Changes made: 
* Created copies of input dataframes before processing them
* Updated docstrings to describe in detail what each method does

File: notebooks/graph_analysis/approaches/graph_helper_functions.ipynb [hunk 2]
```diff
@@ -10,12 +10,13 @@
   },
```
Comment (JamesKunstle) [inline notebooks/graph_analysis/approaches/graph_helper_functions.ipynb:10]: <pre class="ql-syntax" spellcheck="false">Line #4.        This method as input a dataframe assumed to have, at the very least, columns corresponding to repo ids
</pre><p>reword</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/oss-aspen/Rappel/pull/236/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (JamesKunstle) [inline notebooks/graph_analysis/approaches/graph_helper_functions.ipynb:10]: <pre class="ql-syntax" spellcheck="false">Line #44.        # their contributons to a repo, divided by the total number contributons to that repo
</pre><p>this comment is unclear</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/oss-aspen/Rappel/pull/236/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by JamesKunstle** [comment]
@christinaexyou requested a couple of comment modifications. once those are updated I'll go through the analysis very carefully, then I think this'll be ready to merge as additional functionality.

**Comment by christinaexyou** [comment]
@JamesKunstle the following changes were made:

* Added a method named `preprocess` to convert timestamp and contributor id dtypes
* Reworded docstrings in `calc_recency_weights` and `calc_tfidf_weights` methods

**Comment by JamesKunstle** [comment]
@christinaexyou I read through this very carefully- I think your work here is excellent, and I feel good about merging this.

**Pull request merged**